maze_transformer/training/tokenizer.py

```diff
@@ -64,3 +74,94 @@ def as_tokens(
             )
 
         return tokens
+
+
+class HuggingMazeTokenizer(PreTrainedTokenizer):
+    vocab: dict[str, int]  # map of token_ids to strings
+
+    bos_token: str = SPECIAL_TOKENS["path_start"]
+    eos_token: str = SPECIAL_TOKENS["path_end"]
+    pad_token: str = SPECIAL_TOKENS["padding"]
+    unk_token: str = "<UNK>"
+
+    vocab_size: int = 0
+    additional_special_tokens: list[str] = [
+        x for x in SPECIAL_TOKENS.values() if x not in [SPECIAL_TOKENS["padding"]]
+    ]
+
+    # Overwrite class attributes
+    padding_side = "left"
+    truncation_side = "left"  #! strange choice, but it's what we did in pad_sequence
+
+    name_or_path = "maze_tokenizer"
+
+    def __init__(
+        self, cfg: Union["ConfigHolder", "MazeDatasetConfig"], **kwargs
+    ) -> None:
+        # Avoid isinstance() because of circular import
+        if type(cfg).__name__ == "ConfigHolder":
+            cfg = cfg.dataset_cfg
+
+        super().__init__(max_len=cfg.seq_len_max, **kwargs)
+        # We are having to do evil things here
+        vocab = {token: i for i, token in enumerate(cfg.token_arr)}
+        vocab[self.unk_token] = len(vocab)
+        self.vocab = vocab
+
+        self.added_tokens_encoder = vocab
+        self.added_tokens_decoder = {i: token for token, i in vocab.items()}
+
+        self.unique_no_split_tokens = cfg.token_arr
+        self._create_trie(self.unique_no_split_tokens)
+
+        # IDs specified during construction
+        self.bos_token_id = self.added_tokens_encoder[self.bos_token]
+        self.eos_token_id = self.added_tokens_encoder[self.eos_token]
+        self.pad_token_id = self.added_tokens_encoder[self.pad_token]
+
+    def __call__(self, text, **kwargs) -> BatchEncoding:
+        """
+        Tokenizer will take a list of strings and encode each
+        I.e. a single example should be a continuous string
+            "a b c d e f" not ["a", "b", "c", "d", "e", "f"]
+        """
+        try:
+            if isinstance(text, LatticeMaze):
+                text = text.as_tokens(node_token_map=self.vocab)
```

Comment:
`LatticeMaze` doesn't have an `as_tokens` method. Did you mean to use the `MazeTokenizer` here?